sdks/python/src/opik/integrations/anthropic/opik_tracker.py

```diff
@@ -0,0 +1,106 @@
+from typing import Optional
+import logging
+
+import anthropic
+from . import messages_create_decorator
+from . import messages_batch_decorator
+from typing import TypeVar, Dict, Any
+
+AnthropicClient = TypeVar(
+    "AnthropicClient",
+    anthropic.AsyncAnthropic,
+    anthropic.Anthropic,
+    anthropic.AsyncAnthropicBedrock,
+    anthropic.AnthropicBedrock,
+    anthropic.AsyncAnthropicVertex,
+    anthropic.AnthropicVertex,
+)
+
+LOGGER = logging.getLogger(__name__)
+
+
+def track_anthropic(
+    anthropic_client: AnthropicClient,
+    project_name: Optional[str] = None,
+) -> AnthropicClient:
+    """Adds Opik tracking to an Anthropic client.
+
+    Integrates with the following anthropic library objects:
+        * AsyncAnthropic,
+        * Anthropic,
+        * AsyncAnthropicBedrock,
+        * AnthropicBedrock,
+        * AsyncAnthropicVertex,
+        * AnthropicVertex,
+
+    Supported methods (for all classes above) are:
+        * `client.messages.create()`
+        * `client.messages.stream()`
+
+    Can be used within other Opik-tracked functions.
+
+    Args:
+        anthropic_client: An instance of Anthropic client.
+        project_name: The name of the project to log data.
+
+    Returns:
+        Anthropic client with integrated Opik tracking logic.
+    """
+
+    if hasattr(anthropic_client, "opik_tracked"):
+        return anthropic_client
+
+    anthropic_client.opik_tracked = True
+    decorator_factory = messages_create_decorator.AnthropicMessagesCreateDecorator()
+
+    metadata = _extract_metadata_from_client(anthropic_client)
+
+    create_decorator = decorator_factory.track(
+        type="llm",
+        name="anthropic_messages_create",
+        project_name=project_name,
+        metadata=metadata,
+    )
+    stream_decorator = decorator_factory.track(
+        type="llm",
+        name="anthropic_messages_stream",
+        project_name=project_name,
+        metadata=metadata,
+    )
+    batch_create_decorator = messages_batch_decorator.warning_decorator(
+        "At the moment Opik does not support tracking for `client.beta.messages.batches.create` calls",
+        LOGGER,
+    )
+
+    anthropic_client.messages.create = create_decorator(
+        anthropic_client.messages.create
+    )
+    anthropic_client.messages.stream = stream_decorator(
+        anthropic_client.messages.stream
+    )
+    try:
+        anthropic_client.beta.messages.batches.create = batch_create_decorator(
+            anthropic_client.beta.messages.batches.create
+        )
+    except Exception:
+        LOGGER.debug(
+            "Failed to patch messages.batch method. It is likely because it was not implemented in the provided anthropic client",
```

Comment:
```suggestion
            "Failed to patch messages.batches method. It is likely because it was not implemented in the provided anthropic client",
```